XSLT/transpile.xslt

```diff
@@ -34,41 +34,59 @@
   <xsl:include href="modules/functions_module.xslt"/>
   <xsl:include href="assemble-relaxng.xslt"/>
   
+
   <xd:doc>
     <xd:desc><xd:ref name="atop:vMapSchNs"/>: A map of all in-scope namespaces, each with 
     a selected prefix.</xd:desc>
   </xd:doc>
   <xsl:variable name="atop:vMapSchNs" as="map(xs:string, xs:string)" select="atop:get-sch-ns-prefix-map(/)"/>
 
+
   <xd:doc>
-    <xd:desc>The TEI schemaSpec becomes the rng:grammar.</xd:desc>
+    <xd:desc>The TEI &lt;schemaSpec> becomes the RELAX NG &lt;grammar>.</xd:desc>
   </xd:doc>
   <xsl:template match="schemaSpec" as="element(rng:grammar)">
-    <!-- We assume that if there is no @start attribute, the start element is TEI. -->
+    <!-- We assume that if there is no @start attribute, the start element is <TEI>. -->
     <xsl:variable name="vStartElementSpecs" as="element(elementSpec)+"
-                  select="key('atop:elementSpec', if (@start) then tokenize(@start, '\s+') else 'TEI', .)"/>
+                  select="key('atop:elementSpec', if (@start) then tokenize(@start) else 'TEI', .)"/>
     <rng:grammar datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
-      <xsl:variable name="vContext" as="element(schemaSpec)" select="."/>
-      <xsl:for-each select="in-scope-prefixes($vContext)">
-        <xsl:namespace name="{.}" select="namespace-uri-for-prefix(., $vContext)"/>
-      </xsl:for-each>
+      <!--
+	  Make sure to keep namespace declarations, because in RELAX
+	  NG the namespace specified by the prefix of a QName in the
+	  @name attribute takes precedence over the @ns attribute.
+      -->
+      <xsl:copy-of select="namespace-node()"/>
```

Comment:
I would have used ```xsl:sequence select="namespace-node()"```. IIRC this gives the processor room to optimize and deferr creating copies of the nodes in questions.

Reply:
I had forgotten you could do that, if I ever knew. Updated in 7b220f9.